### PR TITLE
cloudscale_server: implement param server_groups

### PR DESCRIPTION
**Pull request by resmo**
##### SUMMARY
Implement new param server_groups, deprecates anti_affinity_with
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

cloudscale_server

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/cloudscale/cloudscale_server.py [hunk 1]
```diff
@@ -378,12 +395,37 @@ def _update_param(self, param_key, server_info, requires_stop=False):
...
+        server_group_ids = []
+        server_groups = self._get('server-groups')
+        for server_group in server_groups:
+            if server_group['uuid'] in server_group_params:
+                server_group_ids.append(server_group['uuid'])
+                server_group_params.remove(server_group['uuid'])
+
+            elif server_group['name'] in server_group_params:
+                server_group_ids.append(server_group['uuid'])
+                server_group_params.remove(server_group['name'])
```
Comment (gaudenz) [inline lib/ansible/modules/cloud/cloudscale/cloudscale_server.py:413]: This has the effect to only add the first server group with this name to the list of server group ids. Is this intended? I'd prefer either warning the user in this case or adding all matching server groups. What do you think?

**Review by gaudenz — CHANGES_REQUESTED** [review]
Thanks for this change. Beside the problem with duplicate names, I think we should extend the test coverage a bit. Maybe it's easiest to rebase this branch onto the cloudscale_server_group branch. Without the `cloudscale_server_group` module this change is pretty useless anyway.

**Comment by resmo** [comment]
ready_for_review

**Comment by resmo** [comment]
rebased

**Pull request merged**